### DISCUSSION TITLE
Generate 3D polycrystalline aggregates from EBSD data

### DISCUSSION THREAD

**Discussion by shuaizhou001** [neperfepx/neper#498] [Q&A] [answered]
Hi @rquey :
Could you explain or make a tutorial on how to generate 3D polycrystalline aggregates from EBSD data?
Now I have the EBSD data (.ctf), this file has three Euler angles and X Y coordinate. What I should do using Neper?

**Comment by rquey** [comment, marked as answer]
[This tutorial](https://neper.info/tutorials/ebsd_process.html) will give you a first idea (the data must be written as a tesr file), and [this tutorial](https://neper.info/tutorials/morpho_tesr_mesh.html) will tell you how to mesh the tesr volume.

If you are missing something, let me know.

**Reply by shuaizhou001** [reply]
I have seen these two tutorials. My confusion is that the EBSD map is 2D, and how to generate a 3D model (Meshing a Polycrystal Image Obtained by Synchrotron X-Ray Diffraction, this tutorial used 3D data).

**Reply by rquey** [reply]
Is your EBSD data one slice or several slices?

If it is several (equally-spaced) slices, you can tentatively write them as a tesr and follow the [tutorial](https://neper.info/tutorials/morpho_tesr_mesh.html).

If it is one slice, it is not enough information to generate a 3D polycrystal. What type of information from the EBSD map do you want to reproduce in the 3D polycrystal?
